### PR TITLE
Populate analytics page w/ new GraphQL endpoint and allow filtering

### LINKED ISSUE #2481: v0: Add top-level metrics to analytics page (Front End)
### Description
We want to make the three most important metrics to users top and center.

### Proposed solution 
This will require both frontend and backend changes.

On the backend, we need to add graphQL queries for the following metrics:
- total tests conducted over the period
- number of positive tests
- positivity rate, expressed as a percentage

Each query needs to be customizable by the dates searched and facilities selected, since users can control those fields.

On the frontend, implement the top three metrics from Victor's design.

### Additional context
It's not clear if we should determine the positivity rate by the total number of positive tests/total tests, or by the _people_ testing positive out of the tested population.

Designs:
analytics sketch

### LINKED ISSUE #2482: v0: Give users ability to filter top-level metrics
### Description
We want users to be able to control how they view their topline metrics.

### Proposed solution 
Create a set of filters for the topline data.

#### Filter 1: Facilities
Create a dropdown that includes all facilities individually, with an organization-level view at the top.

#### Filter 2: Dates
Create a dropdown with the following options:
- last day (24 hours)
- last week (7 days)
- last month (30 days)
- custom date range

There should also be a begin/end date picker. These dates should auto-update depending on the user's dropdown selection.

#### Filter 3: Roles
-all roles
-residents
-staff
-students
-visitors

The default selections for these filters will be the all-facility/organization view,  the last 7 days, and all roles. 
### Additional context
Content for the dropdown text should be reviewed by Rebecca.

Updated comp (9/15/21)

[Old comp](https://user-images.githubusercontent.com/80282552/131922202-d11d975c-7bfe-4896-aa03-4799641b71b4.png)

### PR DESCRIPTION
**Pull request by nickclyde**
## Related Issue or Background Info

- Resolves #2481 
- Resolves #2482

## Changes Proposed

- Consume the `TopLevelDashboardMetricsQuery` in the `Analytics` component
- Allow the user to filter the results
- Wire the `Analytics` component to the `/analytics` route
- Add mocks for story and add tests

## Additional Information

- Removed the `AggregateResultsTable` from the bottom for now until we do some reworking of the GraphQL queries

## Screenshots / Demos

(Disregard the actual data in this demo, I configure the storybook mock to return random numbers)

https://user-images.githubusercontent.com/9121162/134579681-49018d1a-3982-49e3-818a-87d1956354d4.mp4

### UI
- [ ] Any changes to the UI/UX are approved by design 
- [ ] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [ ] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [ ] Database changes are submitted as a separate PR
  - [ ] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [ ] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [ ] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [ ] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

## Cloud
- [ ] DevOps team has been notified if PR requires ops support
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: frontend/src/app/analytics/Analytics.test.tsx [hunk 1]
```diff
@@ -1,13 +1,195 @@
...
+        facilityId: "",
+        startDate: moment(today, EXTERNAL_DATE_FORMAT)
+          .subtract(7, "day")
+          .hour(now.hours())
+          .minute(now.minutes())
+          .toDate(),
+        endDate: moment(today, EXTERNAL_DATE_FORMAT)
+          .hour(now.hours())
+          .minute(now.minutes())
+          .toDate(),
```
Comment (bmaden-usds) [inline frontend/src/app/analytics/Analytics.test.tsx:42]: Interestingly, I think this causes tests that were just submitted not to show up.  Submitting a test, then going to the analytics page wasn't showing the just-submitted test until (what seemed like) the next minute.

**Review by bmaden-usds — COMMENTED** [review]
Thank you, this is looking great!

File: frontend/src/app/analytics/Analytics.tsx [hunk 2]
```diff
@@ -1,55 +1,109 @@
...
+  }
+
+  if (data === undefined) {
+    return <p>Error: Users not found</p>;
+  }
+
+  const totalTests = data.topLevelDashboardMetrics?.totalTestCount || 0;
+  const positiveTests = data.topLevelDashboardMetrics?.positiveTestCount || 0;
+  const negativeTests = totalTests - positiveTests;
+  const positivityRate = (positiveTests / totalTests) * 100;
```
[the remark this answers is not in the document]
Comment (nickclyde) [reply frontend/src/app/analytics/Analytics.tsx:105]: Good catch - it ends up showing as `NaN%`, which isn't ideal - I guess I should make it say `N/A` or something to that effect?

File: frontend/src/app/analytics/Analytics.test.tsx (hunk 1 above)
Comment (nickclyde) [reply frontend/src/app/analytics/Analytics.test.tsx:42]: Wow! Great catch. I'll write a helper function to include the seconds as well as hours and minutes to catch this.

File: frontend/src/app/analytics/Analytics.tsx [hunk 3]
```diff
@@ -1,55 +1,111 @@
...
+  if (loading) {
+    return <LoadingCard />;
+  }
+
+  if (error) {
+    throw error;
+  }
+
+  if (data === undefined) {
+    return <p>Error: Users not found</p>;
```
Comment (emmastephenson) [inline frontend/src/app/analytics/Analytics.tsx:100]: Should this be something like "results not found" instead?

File: frontend/src/app/analytics/Analytics.tsx [hunk 4]
```diff
@@ -147,32 +215,21 @@ export const Analytics = () => {
...
               <div className="card display-flex flex-column flex-align-center">
                 <h2>Negative tests</h2>
-                <h1>26</h1>
+                <h1>{negativeTests}</h1>
                 <p></p>
               </div>
             </div>
             <div className="grid-col-3">
               <div className="card display-flex flex-column flex-align-center">
                 <h2>Positivity rate</h2>
```
Comment (emmastephenson) [inline frontend/src/app/analytics/Analytics.tsx:224]: Can you add a caption for this one explaining that the positivity rate is positive tests/total tests?

**Review by emmastephenson — APPROVED** [review]
Overall looks great, thanks for all your hard work on this! Just some small comments around content.

**Review by bmaden-usds — APPROVED** [review]

**Pull request merged**